### ISSUE TITLE
FP no-useless-cast (S4325): boolean literal as operand

### ISSUE THREAD

**Issue by vilchik-elena** [SonarSource/SonarTS#829]
See #482, Reported in #828 

```ts
const right = {
    error: true as true // cast is flagged as unnecessary
}

typeof right.error // true

const wrong = {
    error: true
}

typeof wrong.error; // boolean
```
